### ISSUE TITLE
Shapes layer with high `scale` values is hard to interact with in the GUI

### ISSUE THREAD

**Issue by andy-sweet** [napari/napari#4268]
## 🐛 Bug

A shapes layer with high `scale` values is hard to modify in the GUI because at some point the blue control points of the selected shape cannot be clearly seen anymore. I tried to click on some of the control points, but I could not manager to move any of them.

Similarly drawing a shape on a shapes layer with a high `scale` value is very difficult, likely for similar reasons.

This seems to be a combination of `scale` values and the current zoom of the camera. I.e. higher scale values and being more zoomed out both contribute to this problem.

Here's a screenshot of what this can look like after hovering (and probably selecting) a shape.

![Screen Shot 2022-03-15 at 2 24 09 PM (2)](https://user-images.githubusercontent.com/2608297/158475800-b19aac41-d4e4-414f-aa15-ecb3342beaf3.png)

## To Reproduce

Here's a code example that causes the issue on my setup.

```python
import napari

shapes_data = [[
    [0, 0],
    [0, 10],
    [10, 10],
    [10, 0],
]]
scale = (1000, 1000)
viewer = napari.Viewer()
shapes = viewer.add_shapes(
    data=shapes_data,
    scale=scale,
)
napari.run()
```

## Expected behavior

The appearance of the blue selection box and control points are independent of the scale of the layer, and maybe the zoom of the camera.

## Environment

```
napari: 0.4.14.dev135+gc8a62b43.d20220314
Platform: macOS-10.16-x86_64-i386-64bit
System: MacOS 11.6.2
Python: 3.9.6 (default, Jul 30 2021, 09:31:09) [Clang 10.0.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.0
SciPy: 1.7.1
Dask: 2021.07.2
VisPy: 0.9.6

OpenGL:
- GL version: 2.1 INTEL-16.5.7
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1440x900, scale 2.0
- screen 2: resolution 2560x1080, scale 1.0

Plugins:
- console: 0.0.4
- scikit-image: 0.4.14.dev135+gc8a62b43.d20220314
```

**Comment by psobolewskiPhD** [comment]
For me this is fixed by https://github.com/napari/napari/pull/5802

**Comment by andy-sweet** [comment]
> For me this is fixed by #5802

Me too!
